source/world_builder/features/continental_plate.cc

```diff
@@ -56,12 +56,44 @@ namespace WorldBuilder
     ContinentalPlate::~ContinentalPlate()
       = default;
 
+    void ContinentalPlate::make_snippet(Parameters &prm)
+    {
+      using namespace rapidjson;
+      Document &declarations = prm.declarations;
+
+      const std::string path = prm.get_full_json_path();
+
+      /*
+      ideally:
+       {
+        "model": "continental plate",
+        "name": "${1:default name}",
+        "dip point":[0.0,0.0],
+        "coordinates": [[0.0,0.0]],
+        "segments": [],
+        "sections": [],
+        "temperature models":[{"model":"uniform", "temperature":600.0}],
+        "composition models":[{"model":"uniform", "compositions": [0], "fractions":[1.0]}]
+       }
+       */
+
+      Pointer((path + "/body").c_str()).Set(declarations,"object");
+      Pointer((path + "/body/model").c_str()).Set(declarations,"continental plate");
+      Pointer((path + "/body/name").c_str()).Set(declarations,"${1:My Plate}");
+      Pointer((path + "/body/coordinates").c_str()).Create(declarations).SetArray();
+      //Pointer((path + "/body/coordinates/0").c_str()).Create(declarations).SetArray();
```

Comment:
Either remove this, or enable it. I am fine either way. Or you could put a 0.0,0.0 coordinate in there.

Reply:
Yeah, one problem is that if you add [] or [0,0], rapidjson ends up putting every single bracket on a single line, which then literally appears when you insert the template. I see the following options:
1. Leave coordinates empty as it is right now.
2. Modify rapidjson to influence the formatting.
3. Insert the snippets manually into the json schema with a script instead of directly in code. That way you can control precisely what formatting you would like.

Preferences?

Reply:
ah, sorry for the late reply. I somehow missed this. Let's leave it as is now. Then we can get it in as part of 0.6 and see from there.